### PR TITLE
Add migration to fix incorrect action IDs for onLoad

### PR DESCRIPTION
**Pull request by sharat87**

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java [hunk 1]
```diff
@@ -827,4 +831,89 @@ public void setExampleAppsInConfig(MongoTemplate mongoTemplate) {
...
+                    for (DslActionDTO actionDTO : actionSet) {
+                        final String actionName = actionDTO.getName();
+                        final Action action = mongoTemplate.findOne(
+                                query(where("pageId").is(page.getId()))
+                                        .addCriteria(where("name").is(actionName)),
+                                Action.class
+                        );
+                        if (action != null) {
+                            actionDTO.setId(action.getId());
+                            actionDTO.setJsonPathKeys(action.getJsonPathKeys());
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:899]: Can we remove setting the json path keys and timeout? This ensures that we are only correcting the erroneous action id in this migration. Thanks!

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java]: Done.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java [hunk 2]
```diff
@@ -827,4 +831,88 @@ public void setExampleAppsInConfig(MongoTemplate mongoTemplate) {
         );
     }
 
+    @ChangeSet(order = "024", id = "clean-action-policies", author = "")
+    public void cleanActionPolicies(MongoTemplate mongoTemplate) {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:835]: Can we rename the function name to correctly represent the functionality? Something along the following lines?
```suggestion
    public void updateErroenousActionIdInPage(MongoTemplate mongoTemplate) {
```

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java [hunk 3]
```diff
@@ -827,4 +831,88 @@ public void setExampleAppsInConfig(MongoTemplate mongoTemplate) {
...
+                for (HashSet<DslActionDTO> actionSet : layoutOnLoadActions) {
+                    for (DslActionDTO actionDTO : actionSet) {
+                        final String actionName = actionDTO.getName();
+                        final Action action = mongoTemplate.findOne(
+                                query(where(FieldName.PAGE_ID).is(page.getId()))
+                                        .addCriteria(where(FieldName.NAME).is(actionName)),
+                                Action.class
+                        );
+                        if (action != null) {
+                            actionDTO.setId(action.getId());
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:898]: Can we add a minor comment here?
```suggestion
                           // update the erroneous action id (template action id) to the cloned action id
                            actionDTO.setId(action.getId());
```

**Review by trishaanand — APPROVED** [review]

**Pull request merged**